projects/ccd-case-ui-toolkit/src/lib/shared/services/form/field-type-sanitiser.ts

```diff
@@ -55,6 +57,39 @@ export class FieldTypeSanitiser {
     });
   }
 
+  public synchronizeCasefieldWithData(caseField: CaseField, data: any): void {
+    for (const field in caseField._value) {
+      for (const id in data[caseField.id]) {
+        if (caseField._value[field]?.id === data[caseField.id][id]?.id) {
+          this.updateFieldValues(caseField._value[field].value, data[caseField.id][id]?.value);
+        }
+      }
+    }
+  }
+
+  public updateFieldValues(caseFieldValue: any, dataValue: any): void {
+    for (const key in dataValue) {
+      if ((typeof caseFieldValue[key] === 'object') && (!isEqual(caseFieldValue[key], dataValue[key]))) {
+        this.updateObjectValue(caseFieldValue[key], dataValue[key]);
+      } else if (typeof caseFieldValue[key] !== 'object') {
+        this.updatePrimitiveValue(caseFieldValue, key, dataValue[key]);
+      }
+    }
+  }
+
+  public updateObjectValue(formattedObject: any, dataObject: any): void {
+    // we only want to update the value if there is a list_items property
```

Comment:
I think you only want to do this for dynamic list objects. You might want to call CaseField.isDynamic, as well as checking if there is a list_items